File: models/pack.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema
const Smartphone = require('./smartphone')

const packSchema = new Schema({
    productType: {type: String, required:true},
    name: {type: String, required: true},
    price: {type: Number, required: true},
    qte: {type: Number, required: true},
    giftName: {type: String, required: true},
    giftQte: {type: Number, required: true},
    smartphoneList: [{
        name: {type: String, required: true},
        price: {type: Number, required: true},
        qte: {type: Number, required: true},
        brand: {type: String, required: true},
        color: {type: String, required: true},
        model: {type: String, require: true},
        quantity: {type: Number, require: true}
    }]
})

const Pack = mongoose.model('pack', packSchema)

module.exports = Pack

File: app/src/main/java/com/tutorial/product_shop/Pack.kt
package com.tutorial.product_shop

class Pack(var smartphoneList:MutableMap<Smartphone,Int>, _id:String, name: String, price: Long, qte: Int): Product(_id, name,
    price, qte) {

    fun addSmartphone(s:Smartphone,qtePhone:Int) = smartphoneList.put(s,qtePhone)
}